compiler/modules/front/codeGen.bal

```diff
@@ -159,6 +162,59 @@ function codeGenStmts(CodeGenContext cx, bir:BasicBlock bb, Scope? initialScope,
     return curBlock;
 }
 
+final readonly & map<bir:OrderOp> RANGE_TO_ORDER = {
+    "...": "<=",
+    "..<": "<"
+};
+
+function codeGenForeachStmt(CodeGenContext cx, bir:BasicBlock startBlock, Scope? scope, ForeachStmt stmt) returns CodeGenError|bir:BasicBlock? {
+    Expr range = stmt.iterable;
+    var { varName, semType} = stmt;
+    if lookup(varName, scope) !== () {
+        return cx.semanticErr(`duplicate declaration of ${varName}`);
+    }
+    if !(range is RangeExpr) {
```

Comment:
The ForeachStmt should have a range member: then you don't need this extra level of indentation.